### PR TITLE
array_of_exprt API

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/util/std_expr.h [hunk 1]
```diff
@@ -1610,11 +1610,6 @@ inline index_exprt &to_index_expr(exprt &expr)
 class array_of_exprt:public unary_exprt
 {
 public:
-  DEPRECATED("use array_of_exprt(what, type) instead")
-  array_of_exprt():unary_exprt(ID_array_of)
-  {
-  }
```
Comment (tautschnig) [inline src/util/std_expr.h]: I guess I don't need to repeat myself.

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 1c8972d570a3e08005f2573911c83f6816f3760f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/104123062

**Review by peterschrammel — APPROVED** [review]
> Each commit message has a non-empty body, explaining why the change was made

I don' see why this box was ticked.

**Review by martin-cs — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: d33773548a6b6991bc16fa311955070b960b3d0a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/105521135

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 728a231a8e37888007698266b764187b0b646452).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/105595168

**Pull request merged**